presto-main/src/main/java/io/prestosql/operator/scalar/JoniRegexpFunctions.java

```diff
@@ -307,4 +308,108 @@ private static void validateGroup(long group, Region region)
             throw new PrestoException(INVALID_FUNCTION_ARGUMENT, format("Pattern has %d groups. Cannot access group %d", region.numRegs - 1, group));
         }
     }
+
+    @ScalarFunction
+    @Description("returns the index of the matched substring")
+    @LiteralParameters("x")
+    @SqlType(StandardTypes.INTEGER)
+    public static long regexpPosition(@SqlType("varchar(x)") Slice source, @SqlType(JoniRegexpType.NAME) JoniRegexp pattern)
+    {
+        return regexpPosition(source, pattern, 1);
+    }
+
+    @ScalarFunction
+    @Description("returns the index of the matched substring starting from the specified position")
+    @LiteralParameters("x")
+    @SqlType(StandardTypes.INTEGER)
+    public static long regexpPosition(@SqlType("varchar(x)") Slice source,
+                                      @SqlType(JoniRegexpType.NAME) JoniRegexp pattern,
+                                      @SqlType(StandardTypes.INTEGER) long start)
+    {
+        return regexpPosition(source, pattern, start, 1);
+    }
+
+    @ScalarFunction
+    @Description("returns the index of the n-th matched substring starting from the specified position")
+    @LiteralParameters("x")
+    @SqlType(StandardTypes.INTEGER)
+    public static long regexpPosition(@SqlType("varchar(x)") Slice source,
+                                      @SqlType(JoniRegexpType.NAME) JoniRegexp pattern,
+                                      @SqlType(StandardTypes.INTEGER) long start,
+                                      @SqlType(StandardTypes.INTEGER) long occurrence)
+    {
+        // start position cannot be smaller than 1
+        if (start < 1) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "start position cannot be smaller than 1");
+        }
+        // occurrence cannot be smaller than 1
+        if (occurrence < 1) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "occurrence cannot be smaller than 1");
+        }
+        // returns -1 if start is greater than the length of source
+        if (start > SliceUtf8.countCodePoints(source)) {
+            return -1;
+        }
+
+        Matcher matcher = pattern.matcher(source.getBytes());
+        long count = 0;
+        // convert char position to byte position
+        // subtract 1 because codePointCount starts from zero
+        int nextStart = SliceUtf8.offsetOfCodePoint(source, (int) start - 1);
+        while (true) {
+            int offset = matcher.search(nextStart, source.length(), Option.DEFAULT);
```

Comment:
Please use loop similar as in `io.prestosql.operator.scalar.JoniRegexpFunctions#regexpReplace(io.airlift.slice.Slice, io.prestosql.type.JoniRegexp, io.airlift.slice.Slice)`